CryptoAnalysis/src/main/java/crypto/analysis/errors/RequiredPredicateError.java

```diff
@@ -33,7 +36,7 @@ public void accept(ErrorVisitor visitor){
 	public String toErrorMarkerString() {
 		String msg = extractedValues.toString();
 		msg += " was not properly generated as ";
-		String predicateName = getContradictedPredicate().getPredName();
+		String predicateName = getContradictedPredicate().stream().map(e -> e.toString()).collect(Collectors.joining(" OR "));
```

Comment:
do  `getPredName` and `toString` yield the same result? Because i wonder why you changed the method call

Reply:
where is that `" OR "` comming from ? Also please use a constant for such important strings.

Reply:
> Do we have a unit test for this issue already? If not, please create one.

Please find the updated tests [here](https://github.com/CROSSINGTUD/CryptoAnalysis/commit/83fc16ea9eccecc4658fce0842d7ec5aa2d1bfc8)

Reply:
> where is that `" OR "` comming from ? Also please use a constant for such important strings.

It is used to generate error statements like below
`Second parameter was not properly generated as generated E C Public Key Parameters(params)  O R generated R S A Private Crt Key Parameters(params)  O R generated R S A Key Parameters(params)`

Reply:
> do `getPredName` and `toString` yield the same result? Because i wonder why you changed the method call

No, they don't. `getPredName` only returns the predicate name. Whereas `toString` returns predicate name with respective parameter, like in above comment. 

Have changed it to getPredName in commit d59e870afdeda7bf5dc13e653c85ded84b6dea51